Source/BestProjectEver/Private/PlayerController/BPE_PlayerController.cpp

```diff
@@ -3,15 +3,69 @@
 
 #include "PlayerController/BPE_PlayerController.h"
 
+#include "Core/GameModes/BPE_GameplayGameMode.h"
 #include "HUD/BPE_HUD.h"
 
+//----------------------------------------------------------------------------------------------------------------------
+ABPE_PlayerController::ABPE_PlayerController()
+{
+	TimeToRespawn = 1.0f;
+}
+
+//----------------------------------------------------------------------------------------------------------------------
 void ABPE_PlayerController::BeginPlay()
 {
 	Super::BeginPlay();
+}
+
+//----------------------------------------------------------------------------------------------------------------------
+void ABPE_PlayerController::OnPossess(APawn* InPawn)
+{
+	Super::OnPossess(InPawn);
+	GetWorldTimerManager().SetTimer(TimerHandle_HUD,this, &ABPE_PlayerController::Client_UpdateHud,
```

Comment:
Arbitrary delays should be avoided if possible. If not, they should be very well documented. I have no idea why you are waiting 0.3 seconds to update the hud since nothing here is telling me anything about it.

Reply:
If the reason for your delay is to allow a variable to replicate, this is not safe and not good practice. Try changing it to a rep notify instead.

Reply:
![image](https://user-images.githubusercontent.com/85327912/173399346-ac515513-1781-4c30-b00d-32390b80aa2b.png)